monolithic_integration_test/src/daphne.rs

```diff
@@ -101,14 +103,53 @@ impl Daphne {
             String::new()
         };
 
-        // Get the test Daphne docker image hash; if necessary, do one-time setup to write the image
-        // to Docker.
-        let image_hash = {
-            let mut image_hash = TEST_DAPHNE_IMAGE_HASH.lock().unwrap();
-            if image_hash.is_none() {
-                *image_hash = Some(load_zstd_compressed_docker_image(TEST_DAPHNE_IMAGE_BYTES));
+        // Get the test Daphne docker image hash; if necessary, do one-time setup to determine the
+        // image name & tag (which may involve writing the image to docker depending on compilation
+        // settings).
+        let (image_name, image_tag) = {
+            let mut image_name_and_tag = TEST_DAPHNE_IMAGE_NAME_AND_TAG.lock().unwrap();
+            if image_name_and_tag.is_none() {
+                let metadata: serde_json::Value =
+                    serde_json::from_slice(TEST_DAPHNE_METADATA_BYTES)
+                        .expect("Couldn't parse Daphne test image metadata");
+                let strategy = metadata["strategy"].as_str().unwrap_or_else(|| {
+                    panic!(
+                        "Metadata strategy field was not a string: {}",
+                        metadata["strategy"]
+                    )
+                });
+
+                *image_name_and_tag = Some(match strategy {
+                    "build" => (
+                        "sha256".to_string(),
+                        load_zstd_compressed_docker_image(TEST_DAPHNE_IMAGE_BYTES),
+                    ),
+
+                    "prebuilt" => (
+                        metadata["image_name"]
+                            .as_str()
+                            .unwrap_or_else(|| {
+                                panic!(
+                                    "Daphne test image metadata image_name field was not a string: {}",
+                                    metadata["image_name"]
+                                )
+                            })
+                            .to_string(),
+                        metadata["image_tag"]
+                            .as_str()
+                            .unwrap_or_else(|| {
+                                panic!(
+                                    "Daphne test image metadata image_tag field was not a string: {}",
+                                    metadata["image_tag"]
+                                )
+                            })
+                            .to_string(),
+                    ),
+
+                    _ => panic!("Unknown Daphne test image build strategy: {strategy:?}"),
```

Comment:
I think we might want a specialized panic message here if the strategy is "skip".